fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/runners/TransactionalRunner.java

```diff
@@ -98,11 +98,37 @@ public TransactionalRunner(@Nonnull FDBDatabase database,
     @SuppressWarnings({"PMD.CloseResource", "PMD.UseTryWithResources"})
     public <T> CompletableFuture<T> runAsync(final boolean clearWeakReadSemantics,
                                              @Nonnull Function<? super FDBRecordContext, CompletableFuture<? extends T>> runnable) {
+        return runAsync(clearWeakReadSemantics, runnable, true);
+    }
+
+    /**
+     * A flavor of the {@link #runAsync(boolean, Function)} method that supports read-only transactions.
+     * @param clearWeakReadSemantics whether to clear the {@link FDBRecordContextConfig#getWeakReadSemantics()} before
+     * creating the transaction. These should be cleared if retrying a transaction, particularly in response to a
+     * conflict, because reusing the old read version would just cause it to re-conflict.
+     * @param runnable some code to run that uses an {@link FDBRecordContext}
+     * @param commitWhenDone if FALSE the transaction will not be committed. If TRUE, behaves the same as described in {@link #runAsync(boolean, Function)}
+     * @param <T> the type of the value returned by the future
+     * @return a future containing the result of the runnable, if successfully committed.
+     * Note: the future will not be {@code null}, but if the runnable returns a future containing {@code null} then
+     * so will the future returned here.
+     */
+    @Nonnull
+    @SuppressWarnings({"PMD.CloseResource", "PMD.UseTryWithResources"})
+    public <T> CompletableFuture<T> runAsync(final boolean clearWeakReadSemantics,
+                                             @Nonnull Function<? super FDBRecordContext, CompletableFuture<? extends T>> runnable,
+                                             boolean commitWhenDone) {
```

Comment:
Looking at the use case, I feel like it would be more readable if this was before the `runnable`.
It may also be better to have it be a different method name (e.g. `runAsyncWithoutCommit`). 

I thought briefly about `readOnlyRunAsync`, but that might imply that it would fail if you tried to do a write.